src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -770,11 +771,39 @@ private async Task AddDocumentsWithPotentialConflictsAsync(IEnumerable<Document>
             {
                 try
                 {
+                    var solutionWithRenameSymbolAnnotations = originalSolution;
+
+                    if (RenameSymbolAnnotation.ShouldAnnotateSymbol(_renameLocationSet.Symbol))
+                    {
+                        // Add the RenameSymbolAnnotation to declarations that will change
+                        foreach (var syntaxReferenceGroup in _renameLocationSet.Symbol.DeclaringSyntaxReferences.GroupBy(r => r.SyntaxTree))
+                        {
+                            var syntaxTree = syntaxReferenceGroup.Key;
+                            var syntaxReferences = syntaxReferenceGroup.AsArray();
+
+                            var document = solutionWithRenameSymbolAnnotations.GetRequiredDocument(syntaxTree);
+                            var documentEditor = await DocumentEditor.CreateAsync(document, _cancellationToken).ConfigureAwait(false);
+
+                            foreach (var syntaxReference in syntaxReferences)
+                            {
+                                var node = await syntaxReference.GetSyntaxAsync(_cancellationToken).ConfigureAwait(false);
+
+                                if (RenameSymbolAnnotation.TryAnnotateNode(node, _renameLocationSet.Symbol, out var annotatedNode))
+                                {
+                                    documentEditor.ReplaceNode(node, annotatedNode);
+                                }
+                            }
+
+                            var annotatedDocument = documentEditor.GetChangedDocument();
+                            solutionWithRenameSymbolAnnotations = annotatedDocument.Project.Solution;
+                        }
+                    }
+
                     foreach (var documentId in documentIdsToRename.ToList())
                     {
                         _cancellationToken.ThrowIfCancellationRequested();
 
-                        var document = originalSolution.GetRequiredDocument(documentId);
+                        var document = solutionWithRenameSymbolAnnotations.GetRequiredDocument(documentId);
```

Comment:
Does anything use originalSolution anymore after this? Or can we just rename something to 'solution' and reassign to that? @dibarbet can attest to the risks of having random locals floating around like this...